cli/tests/pcluster/aws/test_fsx.py

```diff
@@ -124,3 +124,38 @@ def test_describe_volumes(boto3_stubber):
     # Fourth boto3 call after resetting the AWSApi instance. The latest fsx lifecycle should be retrieved from boto3
     AWSApi.reset()
     assert_that(AWSApi.instance().fsx.describe_volumes([volume])[0]["Lifecycle"]).is_equal_to("CREATED")
+
+
+def get_describe_file_caches_mocked_request(file_cache_ids, lifecycle):
+    return MockedBoto3Request(
+        method="describe_file_caches",
+        response={
+            "FileCaches": [{"FileCacheId": file_cache_id, "Lifecycle": lifecycle} for file_cache_id in file_cache_ids]
+        },
+        expected_params={"FileCacheIds": file_cache_ids},
+    )
+
+
+def test_describe_file_caches(boto3_stubber):
+    file_cache_id = "fc-12345678"
+    additional_file_cache_id = "fc-123456789012345678"
+    # The first mocked request and the third are about the same File Cache. However, the lifecycle of the FC changes
+    # from CREATING to AVAILABLE. The second mocked request is about another FC
+    mocked_requests = [
+        get_describe_file_caches_mocked_request([file_cache_id], "CREATING"),
+        get_describe_file_caches_mocked_request([additional_file_cache_id], "AVAILABLE"),
+        get_describe_file_caches_mocked_request([file_cache_id], "AVAILABLE"),
+    ]
+    boto3_stubber("fsx", mocked_requests)
+    assert_that(AWSApi.instance().fsx.describe_file_caches([file_cache_id])[0]["Lifecycle"]).is_equal_to("CREATING")
+
+    # Second boto3 call with more FileCaches. The FC already cached should not be included in the boto3 call.
+    response = AWSApi.instance().fsx.describe_file_caches([file_cache_id, additional_file_cache_id])
+    assert_that(response).is_length(2)
```

Comment:
Not clear to me why this assertion should prove that the cached file cache is not included in the boto3 call.
If your goal is to verify that a describe call is not execute for a cached FC, then you need to assert that the underlying boto3 call has not been called with that id as parameter.

Reply:
I think this would be more complicated in terms of checking for underlying boto3 call. But I can check for the fc_cache contains values before and after.